module/VuFind/src/VuFind/Search/Base/Results.php

```diff
@@ -625,6 +632,38 @@ public function setExtraData(array $data): void
         }
     }
 
+    /**
+     * Get optional parameters for the search.
+     *
+     * The optional parameters mechanism is used for storing search parameters
+     * that may be useful for reproducing the full context of a user's search,
+     * but which do not impact the contents of the search result set -- e.g. page number.
+     * Contents must be serializable. It is recommended to make the array as small
+     * as possible.
+     *
+     * @return array
+     */
+    public function getOptionalParameters(): array
+    {
+        return $this->optionalParameters;
```

Comment:
Is there a reason that we need to store optional parameters in a specific property here? Could we instead just proxy the relevant settings in the params object, e.g. have getOptionalParameters return

```
[
    'page' => $this->getParams()->getPage(),
    'limit' => $this->getParams()->getLimit(),
]
```

and have the setter proxy the appropriate params setters when the incoming array contains appropriate values?

And if we do it that way, should these methods actually go on the Params class rather than the Results class, or might there be a reason to keep them tied to results?

I realize this may all be by design, especially if doing it "by proxy" causes problems with search deduplication, etc. I can't remember exactly how that works, so depending on the existing approach, my idea might or might not be problematic. Just wondering if this has been considered, because I think this approach could be a bit confusing -- i.e. why are we storing page and limit in two different places?

Reply:
Hmm, good point, I'll try and see what happens!